### PR TITLE
Localization Replay Viewer

### PR DESCRIPTION
**Pull request by knoellle**
## Introduced Changes

Adds a new tool at `tools/localizer` which can load and replay log files generated by #468.
It can be controlled the same way as the behavior simulator but only provides the information stored in the log file augmented with line correspondences.
A map layer to show line correspondences is also added to twix.
Tool code is hacky and optimized to reduce code changes elsewhere.

Fixes #

## Ideas for Next Iterations (Not This PR)

Proper replay & viewer

## How to Test

```
cd tools/localizer
cargo run <path-to-localization-log.bincode>
```
- Open twix, map panel, behavior simulator panel, connect to localhost, turn on line correspondences.
- Gaze upon the robots path and be delighted.

### CODE REVIEW

File: tools/twix/src/panels/map/layers/line_correspondences.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    nao::Nao, panels::map::layer::Layer, twix_painter::TwixPainter, value_buffer::ValueBuffer,
+};
+
+pub struct LineCorrespondences {
+    lines_in_robot_bottom: ValueBuffer,
+    lines_in_robot_top: ValueBuffer,
+}
+
+impl Layer for LineCorrespondences {
+    const NAME: &'static str = "Line Associations";
```
Comment (h3ndrk) [inline tools/twix/src/panels/map/layers/line_correspondences.rs:19]:
```suggestion
    const NAME: &'static str = "Line Correspondences";
```

File: tools/localizer/src/main.rs
Comment (h3ndrk) [inline tools/localizer/src/main.rs:1]: Where can I find a localization log file to test this with?

**Review by h3ndrk — CHANGES_REQUESTED** [review]

**Review by h3ndrk — DISMISSED** [review]
Wörks

File: tools/twix/src/panels/map/layers/line_correspondences.rs (hunk 1 above)
Comment (knoellle) [reply tools/twix/src/panels/map/layers/line_correspondences.rs]: Done

**Review by h3ndrk — APPROVED** [review]

**Pull request merged**